cranelift/codegen/src/isa/s390x/inst/mod.rs

```diff
@@ -2121,7 +2114,7 @@ impl Inst {
             }
             &Inst::FpuCMov64 { rd, cond, ri, rm } => {
                 let (rd, rd_fpr) = pretty_print_fpr(rd.to_reg(), allocs);
-                let _ri = allocs.next(ri);
+                let _ri = ri;
```

Comment:
Minor, but could these assignments be removed too?

Reply:
Yes, there are several kinds of assignments that should still be removed, but they're more difficult to do mechanically so I wanted to save them for a separate PR.
- `let _ri = ri;` or `let _ = u.vreg;` has the side effect of suppressing warnings that the value on the right-hand side is not used. This is not an idiomatic way to do it, but simply deleting the assignment will introduce warnings, which breaks CI unless we do something about them.
- `let tmp = *tmp;` dereferences a borrow of the `tmp` field and re-binds the name `tmp` to a copy of the field. Rust has syntax for doing this directly in the pattern match that bound the name `tmp` in the first place, which I'd prefer to use.
- `let ri_hi = ri.hi;` or `let r = rn;` could be deleted as long as all the uses of the left-hand side are replaced by the right-hand side.

The changes in this PR are only the ones I could script with `sed` and the other tools I was using, and I want to stop at this point so this review is pretty mechanical too.